falcon/http_error.py

```diff
@@ -46,7 +46,10 @@ class HTTPError(Exception):
         error in a future version of falcon.
 
     Args:
-        status (str): HTTP status code and text, such as "400 Bad Request"
+        status (Union[str,int]): HTTP status code or line (e.g.,
```

Comment:
we don't seem to handle the case of int/httpstatus here, like in `self.title = title or status`

Reply:
Added tests and a fix for this.